# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1769778780 -39600
# Parent  b3d379233b25c331d3c74172e555cf620374c164
Branding 8: Change sign.py and sign_file.py to use Certum certificate.

diff --git a/browser/installer/windows/sign.py b/browser/installer/windows/sign.py
--- a/browser/installer/windows/sign.py
+++ b/browser/installer/windows/sign.py
@@ -1,16 +1,19 @@
 import subprocess, sys
 
 p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
-    "Get-ChildItem -Path ..\\..\\..\\dist\\bin -Filter *.exe -File | Where-Object { $_.Name -in @(\"betterbird.exe\", \"pingsender.exe\", \"plugin-container.exe\", \"rnp-cli.exe\", \"rnpkeys.exe\", \"WSEnable.exe\") } | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\dist\\bin\\$_ -Certificate (Get-ChildItem -Path Cert:\\CurrentUser\\My\\ -CodeSigningCert) -TimestampServer http://timestamp.digicert.com } "],
+    f"$cert = Get-ChildItem Cert:\\CurrentUser\\My\\ | Where-Object {{ $_.Thumbprint -eq 'c39501225a337c86f91e3ba7911acb06ff48acf3' }}; "
+    "Get-ChildItem -Path ..\\..\\..\\dist\\bin -Filter *.exe -File | Where-Object { $_.Name -in @(\"betterbird.exe\", \"pingsender.exe\", \"plugin-container.exe\", \"rnp-cli.exe\", \"rnpkeys.exe\", \"WSEnable.exe\") } | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\dist\\bin\\$_ -Certificate $cert -TimestampServer http://time.certum.pl } "],
     stdout=sys.stdout)
 p.communicate()
 
 p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
-    "Get-ChildItem -Path ..\\..\\..\\dist\\bin\\uninstall -Filter helper.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\dist\\bin\\uninstall\\$_ -Certificate (Get-ChildItem -Path Cert:\\CurrentUser\\My\\ -CodeSigningCert) -TimestampServer http://timestamp.digicert.com } "],
+    f"$cert = Get-ChildItem Cert:\\CurrentUser\\My\\ | Where-Object {{ $_.Thumbprint -eq 'c39501225a337c86f91e3ba7911acb06ff48acf3' }}; "
+    "Get-ChildItem -Path ..\\..\\..\\dist\\bin\\uninstall -Filter helper.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\dist\\bin\\uninstall\\$_ -Certificate $cert -TimestampServer http://time.certum.pl } "],
     stdout=sys.stdout)
 p.communicate()
 
 p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
-    "Get-ChildItem -Path ..\\..\\..\\dist\\bin -Filter *.dll -File | Where-Object { $_.Name -notlike \"vcruntime140*\" -and $_.Name -notlike \"msvcp140.*\" -and $_.Name -notlike \"api-ms-win*\" -and $_.Name -notlike \"d3dcompiler_47.*\" } | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\dist\\bin\\$_ -Certificate (Get-ChildItem -Path Cert:\\CurrentUser\\My\\ -CodeSigningCert) -TimestampServer http://timestamp.digicert.com } "],
+    f"$cert = Get-ChildItem Cert:\\CurrentUser\\My\\ | Where-Object {{ $_.Thumbprint -eq 'c39501225a337c86f91e3ba7911acb06ff48acf3' }}; "
+    "Get-ChildItem -Path ..\\..\\..\\dist\\bin -Filter *.dll -File | Where-Object { $_.Name -notlike \"vcruntime140*\" -and $_.Name -notlike \"msvcp140.*\" -and $_.Name -notlike \"api-ms-win*\" -and $_.Name -notlike \"d3dcompiler_47.*\" } | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\dist\\bin\\$_ -Certificate $cert -TimestampServer http://time.certum.pl } "],
     stdout=sys.stdout)
 p.communicate()
diff --git a/browser/installer/windows/sign_file.py b/browser/installer/windows/sign_file.py
--- a/browser/installer/windows/sign_file.py
+++ b/browser/installer/windows/sign_file.py
@@ -5,9 +5,10 @@ parser.add_argument("--input", required=
 args = parser.parse_args()
 
 cmd = [
     r"C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe",
-    f"Set-AuthenticodeSignature -FilePath \"{args.input}\" -Certificate (Get-ChildItem -Path Cert:\\CurrentUser\\My\\ -CodeSigningCert) -TimestampServer http://timestamp.digicert.com"
+    f"$cert = Get-ChildItem Cert:\\CurrentUser\\My\\ | Where-Object {{ $_.Thumbprint -eq 'c39501225a337c86f91e3ba7911acb06ff48acf3' }}; "
+    f"Set-AuthenticodeSignature -FilePath \"{args.input}\" -Certificate $cert -TimestampServer http://time.certum.pl"
 ]
 
 p = subprocess.Popen(cmd, stdout=sys.stdout, stderr=sys.stderr)
 p.communicate()
